lua/neo-tree/command/init.lua

```diff
@@ -72,7 +72,8 @@ M.execute = function(args)
 
   -- Now get the correct state
   local state
-  if args.position == "current" then
+  local requested_position = args.position or nt.config[args.source].window.position
+  if requested_position == "current" then
```

Comment:
Just to clarify, I think that these two lines are the bug fix, and everything else should be reverted to the original code and left unchanged.

Reply:
Actually with just that fix in place it leaves the netrw hijack broken.

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/039aff445280dd4ffce3c4b97e25d7962c0bcfd4/lua/neo-tree/setup/netrw.lua#L83-L95

Logging here shows that `should_open_current` is false and control goes to the `hijack_netrw: opening default` block.

Reply:
OH! I get it now. The problem is that the `get_position` method was not updated when it was refactored out of the `setup/init/lua` module into the the separate `netrw.lua` module, so it has just been broken the whole time.

What you did works but it's not correct. The `netrw.get_position` method should either be changed to:
```lua
local get_position = function(source_name)
  local nt = require("neo-tree")
  local pos = utils.get_value(nt, "config." .. source_name .. ".window.position", "left", true)
  return pos
end
```
or that function should just be completely removed and the above code inlined to the one place that it is actually called from.

Reply:
I've reverted the previous commit. With the snippet above, the netrw hijack won't respect the global `config.window.position` when the `config.filesystem.window.position` is unset and `netrw_hijack_behaviour = open_default`. Is this the expected behaviour?

Reply:
> config.window.position when the config.filesystem.window.position is unset and netrw_hijack_behaviour = open_default

That means there is a problem in the setup function that needs to be fixed. When `config.filesystem.window.position` should never be unset, if the user doesn't specify a value then it should already be set to the global value before any code runs. If that is not working correctly, it could be causing other bugs as well and should be fixed.

> Also is it preferable to overwrite history to remove a bad commit in a PR or just revert and push that?

It doesn't matter, whatever you think is easiest to follow. I tend to keep adding commits and only force push when rebasing to include changes merged to the main branch. In the end it will be squashed to a single commit when it is merged.

Reply:
Apologies, it actually was set correctly and getting the global config win position wasn't necessary. I've pushed another commit to fix that. Thank you for hand-holding my way through this :)